.travis.yml

```diff
@@ -69,39 +72,30 @@ env:
 # Install system dependencies, namely a very barebones ROS setup.
 before_install:
   - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+  - sudo rm /var/lib/dpkg/lock
+  - sudo dpkg --configure -a
+  - sudo apt-get update
+  - sudo apt-get install ros-$ROS_DISTRO-desktop-full python-catkin-pkg python-wstool python-catkin-tools
```

Comment:
desktop-full takes a *long* time to install, so we should avoid it if possible. What packages have you found missing with the example (i.e. very basic) repository. Does the example not compile with ros-$ROS_DISTRO-catkin only?

Reply:
I used the same configuration I need for my ROS package. Maybe, it is not necessary for your example. You should try.